hoot-core/src/main/cpp/hoot/core/ops/UnconnectedWaySnapper.cpp

```diff
@@ -561,73 +559,118 @@ bool UnconnectedWaySnapper::_snapUnconnectedNodeToWay(NodePtr nodeToSnap)
     // for each neighboring way
     WayPtr wayToSnapTo = _map->getWay((*waysToSnapToItr).getId());
     LOG_VART(wayToSnapTo);
-    const std::vector<long>& wayNodeIdsToSnapTo = wayToSnapTo->getNodeIds();
-
-    // Make sure the way we're trying to snap to doesn't already contain the node we're trying to
-    // snap to it.
-    const bool wayToSnapToContainsNodeToSnap =
-      std::find(wayNodeIdsToSnapTo.begin(), wayNodeIdsToSnapTo.end(), nodeToSnap->getId()) !=
-      wayNodeIdsToSnapTo.end();
-    LOG_VART(wayToSnapToContainsNodeToSnap);
-    if (!wayToSnapToContainsNodeToSnap)
+    if (_snapUnconnectedNodeToWay(nodeToSnap, wayToSnapTo))
     {
-      // find the closest coord on the neighboring way to our input node
-      double shortestDistanceFromNodeToSnapToWayCoord = DBL_MAX;
-      const geos::geom::Coordinate closestWayToSnapToCoord =
-        OsmUtils::closestWayCoordToNode(nodeToSnap, wayToSnapTo,
-          shortestDistanceFromNodeToSnapToWayCoord, _snapToWayDiscretizationSpacing, _map);
-
-      // This check of the calculated distance being less than the allowed snap distance should not
-      // be necessary, but it is for now.  For some reason, neighbors are occasionally being
-      // returned at longer distances away than expected.
-      if (/*shortestDistance != DBL_MAX &&*/
-          shortestDistanceFromNodeToSnapToWayCoord <= _maxSnapDistance)
-      { 
-        // snap the node to the way
-
-        // figure out where on the target way to insert our node being snapped
-        const int nodeToSnapInsertIndex = _getNodeToSnapWayInsertIndex(nodeToSnap, wayToSnapTo);
-
-        if (Log::getInstance().getLevel() <= Log::Debug)
-        {
-          // This projection could be expensive, so do it during debugging only.
-          const geos::geom::Coordinate wgs84Coord =
-            MapProjector::project(
-              closestWayToSnapToCoord, _map->getProjection(),
-              MapProjector::createWgs84Projection());
-          LOG_TRACE(
-            "Snapping way node: " << nodeToSnap->getId() << " to coord: " <<
-            closestWayToSnapToCoord.toString() << " (wgs84: " << wgs84Coord.toString() <<
-            ") and inserting at index: " << nodeToSnapInsertIndex);
-        }
```

Comment:
Everything here was removed and put into a function below, with no changes so that it could be reused.